Rust version compatibility

Both R-hub builder and R-universe builder (`ubuntu:jammy`) failed installation with the following error after using `prql-compiler` 0.4.0.
https://github.com/r-universe/eitsupi/actions/runs/3941624440/jobs/6752975644#step:3:802

```log
  error[E0658]: `let...else` statements are unstable
     --> /tmp/RtmpnXSnJF/R.INSTALLfa93bec8912/prqlr/src/.cargo/registry/src/github.com-1ecc6299db9ec823/prql-compiler-0.4.0/src/semantic/lowering.rs:453:13
      |
  453 | /             let pl::ExprKind::All { except, .. } = expr.kind else {
  454 | |                 // base case
  455 | |                 r.push(self.declare_as_column(expr, is_aggregation)?);
  456 | |                 continue;
  457 | |             };
      | |______________^
      |
      = note: see issue #87335 <https://github.com/rust-lang/rust/issues/87335> for more information
```

#38 does not seem enough for R-universe.

```log
   	cargo build --lib --release --manifest-path=./rust/Cargo.toml --target-dir ./rust/target
      Updating git repository `https://github.com/extendr/extendr`
      Updating crates.io index
      Updating git repository `https://github.com/PRQL/prql`
  error: failed to get `prql-compiler` as a dependency of package `prqlr v0.1.0 (/tmp/RtmpilNWCq/Rbuild5ef55c3be2a/prqlr/src/rust)`
  
  Caused by:
    failed to load source for dependency `prql-compiler`
  
  Caused by:
    Unable to update https://github.com/PRQL/prql?rev=1b4217c01c590c51a3e80c8a11b28f78f109a110#1b4217c0
  
  Caused by:
    object not found - no match for id (1b4217c01c590c51a3e80c8a11b28f78f109a110); class=Odb (9); code=NotFound (-3)
  make: *** [Makevars:16: rust/target/release/libprqlr.a] Error 101
  ERROR: compilation failed for package ‘prqlr’
```

--- a/src/rust/Cargo.toml
+++ b/src/rust/Cargo.toml
@@ -8,8 +8,9 @@ crate-type = ['staticlib']
 name = "prqlr"
 
 [dependencies]
-extendr-api = { git = "https://github.com/extendr/extendr", rev = "f511254b1be180528a1711e4c87473966a1010d0" }
 # extendr-api = "0.3.1"
-prql-compiler = { git = "https://github.com/PRQL/prql", rev = "996a783b56186c3711e065a5b347e6489155085e", default-features = false }
+extendr-api = { git = "https://github.com/extendr/extendr", rev = "f511254b1be180528a1711e4c87473966a1010d0" }
 # prql-compiler = { version = "0.4.0", default-features = false }
+# prql-compiler 0.4.0 is not compatible with Rust 1.64, so a slightly modified version is installed from the fork. https://github.com/PRQL/prql/pull/1561
+prql-compiler = { git = "https://github.com/eitsupi/prql", rev = "996a783b56186c3711e065a5b347e6489155085e", default-features = false }
 anyhow = "1.0.68"

